x/staking/types/delegation.go

```diff
@@ -335,3 +343,128 @@ func (d Redelegations) String() (out string) {
 	}
 	return strings.TrimSpace(out)
 }
+
+// ----------------------------------------------------------------------------
+// Client Types
+
+// DelegationResponse is equivalent to Delegation except that it contains a balance
+// in addition to shares which is more suitable for client responses.
+type DelegationResponse struct {
+	Delegation
+	Balance sdk.Int `json:"balance"`
+}
+
+func NewDelegationResp(d sdk.AccAddress, v sdk.ValAddress, s sdk.Dec, b sdk.Int) DelegationResponse {
+	return DelegationResponse{NewDelegation(d, v, s), b}
```

Comment:
```suggestion
	return DelegationResponse{NewDelegation(delegatorAddr, validatorAddr, shares), balance}
```

Reply:
The function body is short and concise. No need for verbose args: https://dave.cheney.net/practical-go/presentations/qcon-china.html#_identifier_length

Reply:
I agree @alexanderbez - godoc comment would be nice though

Reply:
when the `type` matches the variable name I agree (like we do with the keepers), but in this case for example `shares` and `balances` are `Dec` and `Int` respectively

Reply:
I don't see how expanding the variable names would help here? Again, it's concise.

Reply:
> The function body is short and concise. No need for verbose args: https://dave.cheney.net/practical-go/presentations/qcon-china.html#_identifier_length

Again, that applies for examples like `p Person` or `i index`. In the function above I agree that the naming is concise, but not descriptive for the reasons I stated in my previous comment.

I just don't want us to end up being known for our naming as MakerDAO is: https://makerdao.com/purple/#sec-1-3, https://www.reddit.com/r/MakerDAO/comments/bij1dj/makerdao_variables_that_are_developer_friendly/

Reply:
I think it's a matter of brevity and developer convenience VS human readability balance. I tend to appreciate brevity a lot, as long as functions syntax and semantic are documented. IMHO a good compromise would be to keep variable names short for conciseness's sake _and_ provide a descriptive godoc comment.